shared/src/Piipan.Shared/Logging/RequestLoggingMiddleware.cs

```
@@ -14,12 +15,19 @@ public RequestLoggingMiddleware(RequestDelegate next)
             _next = next;
         }
 
-        public async Task InvokeAsync(HttpContext context, 
+        public async Task InvokeAsync(HttpContext context,
             ILogger<RequestLoggingMiddleware> logger,
             IClaimsProvider claimsProvider)
         {
-            logger.LogInformation($"{claimsProvider.GetEmail(context.User)} {context.Request.Method} {context.Request.Path}");
-            await _next(context);
+            logger.LogInformation($"Request {context.TraceIdentifier}: {claimsProvider.GetEmail(context.User)} {context.Request.Method} {context.Request.Path}");
+            try
+            {
+                await _next(context);
+            }
+            catch (Exception ex)
+            {
+                logger.LogError(ex, $"Request {context.TraceIdentifier} Exception: {ex.Message}");

```

Question: This exception is now getting swallowed. Is that an intentional change?

---

It was, but I'm wondering now if it should. Maybe it shouldn't, so that the redirect to the error page still happens. What are your thoughts?

---

It's an interesting question. My inclination would be to to say no. If an exception is being thrown (and not caught) by the API, I wouldn't necessarily place exception handling responsibility in the Logging middleware. Now, if we were registering exception handling middleware, e.g. something like ExceptionHandlingMiddleware, then swallowing the exceptions might be ok. That might be something we should consider in a future ticket

---

Sure, that makes a lot of sense. I'll throw it. I think mainly this is going to be for the initial page rendering (not API calls) since those should be well handled now, but I'll let Startup take care of it, and I'll add an ExceptionHandlingMiddleware to the list of additional use cases not covered by this work.